tools/link.py

```diff
@@ -2573,9 +2573,10 @@ def generate_traditional_runtime_html(target, options, js_target, target_basenam
 
   shell = do_replace(shell, '{{{ SCRIPT }}}', script.replacement())
   shell = shell.replace('{{{ SHELL_CSS }}}', utils.read_file(utils.path_from_root('src/shell.css')))
-  logo_filename = utils.path_from_root('media/powered_by_logo_shell.png')
-  logo_b64 = base64_encode(logo_filename)
-  shell = shell.replace('{{{ SHELL_LOGO }}}', f'<img id="emscripten_logo" src="data:image/png;base64,{logo_b64}">')
+  if '{{{ SHELL_LOGO }}}' in shell:
+    logo_filename = utils.path_from_root('media/powered_by_logo_shell.png')
+    logo_b64 = base64_encode(logo_filename)
+    shell = shell.replace('{{{ SHELL_LOGO }}}', f'<img id="emscripten_logo" src="data:image/png;base64,{logo_b64}">')
```

Comment:
Is this worth it?   If so, perhaps we should do the same for `SHELL_CSS` above?

Reply:
Actually, since this was changed in the other PR that landed earlier today to avoid having to read the file in, it might be fine now? Before, the Python script would fail since it couldn't read the file so it was necessary, but I think now if it's just a path it should work?

As for the CSS though, since that's being read in, I did add that one as well.

Reply:
The change I made today doesn't stop the file being read in.. it just modes the call to read into `base64_encode`.

The `logo_filename` should always exist in emscripten so I don't think it should ever fail to read that file.